src/scp/test/SCPTests.cpp

```diff
@@ -2452,6 +2510,53 @@ TEST_CASE("ballot protocol core3", "[scp][ballotprotocol]")
             }
         }
     }
+    SECTION("prepared A1 with timeout")
+    {
+        // starts with bValue (smallest)
+        REQUIRE(scp.bumpState(0, bValue));
+        REQUIRE(scp.mEnvs.size() == 1);
+
+        // setup
+        recvQuorumChecks(makePrepareGen(qSetHash, A1, &A1, 0, 1), false, false);
+        REQUIRE(scp.mEnvs.size() == 2);
+        verifyPrepare(scp.mEnvs[1], v0SecretKey, qSetHash0, 0, A1, &A1, 1, 1);
+
+        // now, receive bumped votes
+        recvQuorumChecks(makePrepareGen(qSetHash, A2, &B2, 0, 1, &A1), true,
+                         true);
+        REQUIRE(scp.mEnvs.size() == 3);
+        // p=B2, p'=A1 (1)
+        // computed_h = B2 (2)
+        //   does not update h as b < computed_h
+        // v-blocking ahead -> b = computed_h = B2 (9)
+        // h = B2 (2) (now possible)
+        // c = 0 (1)
+        verifyPrepare(scp.mEnvs[2], v0SecretKey, qSetHash0, 0, B2, &A2, 0, 2,
+                      &B2);
+    }
+    SECTION("third party timing")
```

Comment:
also not quite sure what this name means

Reply:
I don't have a good name for this, I called it third party as it's a watcher node that doesn't have itself in its quorum set. I guess I'm going to rename it to something else